src/cantools/database/can/c_source.py

```diff
@@ -1240,6 +1241,9 @@ def _generate_encode_decode(cg_signal: "CodeGenSignal", use_float: bool) -> tupl
         encoding = f'value / {scale_literal}'
         decoding = f'({floating_point_type})value * {scale_literal}'
 
+    if not cg_signal.signal.is_float and use_round:
```

Comment:
Hey! I'm not sure here should it be `cg_signal.signal.is_float` or `cg_signal.signal.conversion.is_float`...
@eerimoq Could you please advise, and if possible review this PR? Thank You in advance!